File: Queue/ArrayQueue.rb
class ArrayQueue
    def initialize
        @data = []
    end
    
    def empty?
        return @data.empty?
    end
    
    def peek
        return @data.first
    end
    
    def enqueue(data)
        @data << data
    end
    
    def dequeue
        return @data.shift
    end
end

File: Queue/ArrayQueue.py
class ArrayQueue:
    def __init__(self):
        self.__data = []

    def isempty(self):
        return self.__data == []

    def search(self, key):
        return self.__data.index(key) if key in self.__data else None

    def peek(self):
        return self.__data[0] if not self.isempty() else None

    def enqueue(self, data):
        self.__data.append(data)

    def dequeue(self):
        return self.__data.pop(0) if not self.isempty() else None
